src/plugins/validation/2and3/semantic-validators/responses.js

```diff
@@ -19,23 +19,60 @@ module.exports.validate = function({ jsSpec, isOAS3 }, config) {
       each(obj, (response, responseKey) => {
         if (isOAS3) {
           each(response.content, (mediaType, mediaTypeKey) => {
-            const hasInlineSchema = mediaType.schema && !mediaType.schema.$ref;
+            const combinedSchemaTypes = ['allOf', 'oneOf', 'anyOf'];
+            const hasOfProperty =
+              mediaType.schema.allOf ||
+              mediaType.schema.anyOf ||
+              mediaType.schema.oneOf;
```

Comment:
First of all, this is very picky, but let's rename `hasOfProperty` to `hasCombinedSchema`.

Second of all, this will break if `mediaType.schema` is undefined. Just move this variable assignment to be within the if statement below (`mediaType.schema`).